libs/langchain/pyproject.toml

```diff
@@ -37,7 +37,7 @@ tiktoken = {version = ">=0.3.2,<0.6.0", optional = true, python=">=3.9"}
 qdrant-client = {version = "^1.3.1", optional = true, python = ">=3.8.1,<3.12"}
 dataclasses-json = ">= 0.5.7, < 0.7"
 cohere = {version = "^4", optional = true}
-openai = {version = "<2", optional = true}
+openai = {version = ">=1.14.3, <2", optional = true}
```

Comment:
openai is a widely used dependency. Could we relax the constraint here instead and make the source code work with both old and new versions of openai?